internal/langserver/handlers/tfschema/candidates.go

```diff
@@ -60,10 +60,24 @@ func valueCandidates(values []string, r lsp.Range, isOrdered bool) []lsp.Complet
 	return candidates
 }
 
-func typeCandidates(prefix *string, r lsp.Range) []lsp.CompletionItem {
+func resourceTypeCandidates(prefix *string, r lsp.Range) []lsp.CompletionItem {
+	return typeCandidates(prefix, r, false)
+}
+
+func dataSourceTypeCandidates(prefix *string, r lsp.Range) []lsp.CompletionItem {
+	return typeCandidates(prefix, r, true)
+}
+
+func typeCandidates(prefix *string, r lsp.Range, allowReadOnly bool) []lsp.CompletionItem {
 	candidates := make([]lsp.CompletionItem, 0)
 	if prefix == nil || !strings.Contains(*prefix, "@") {
-		for resourceType := range azure.GetAzureSchema().Resources {
+		for resourceType, resource := range azure.GetAzureSchema().Resources {
+			if len(resource.Definitions) > 0 {
```

Comment:
Will there be case that the length > 1?

Reply:
Yes, but it's not necessary to check all definitions, it's enough to take the first one.